tests/devices/qubit/test_sampling.py

```diff
@@ -252,3 +252,99 @@ def test_approximate_expval_measure(self):
 
         assert result != 0
         assert np.allclose(result, 0, atol=0.03)
+
+    @pytest.mark.parametrize(
+        "shots, total_copies",
+        [
+            [(100,), 1],
+            [((100, 1),), 1],
+            [((100, 2),), 2],
+            [(100, 100), 2],
+            [(100, 200), 2],
+            [(100, 100, 200), 3],
+            [(200, (100, 2)), 3],
+        ],
+    )
+    def test_sample_measure_shot_vector(self, shots, total_copies):
+        """Test that a sample measurement with shot vectors works as expected"""
+        state = qml.math.array(two_qubit_state)
+        shots = qml.measurements.Shots(shots)
+        mp = qml.sample(wires=range(2))
+
+        result = measure_with_samples(mp, state, shots=shots)
+
+        if total_copies == 1:
+            assert isinstance(result, np.ndarray)
+            result = (result,)
+
+        assert isinstance(result, tuple)
+        assert len(result) == total_copies
+
+        for res, sh in zip(result, shots):
+            assert res.shape == (sh, 2)
+            assert res.dtype == np.bool8
+            assert all(qml.math.allequal(s, [0, 1]) or qml.math.allequal(s, [1, 0]) for s in res)
+
+    @pytest.mark.parametrize(
+        "shots, total_copies",
+        [
+            [(100,), 1],
+            [((100, 1),), 1],
+            [((100, 2),), 2],
+            [(100, 100), 2],
+            [(100, 200), 2],
+            [(100, 100, 200), 3],
+            [(200, (100, 2)), 3],
+        ],
+    )
+    def test_prob_measure_shot_vector(self, shots, total_copies):
+        """Test that a probability measurement with shot vectors works as expected"""
+        state = qml.math.array(two_qubit_state)
+        shots = qml.measurements.Shots(shots)
+        mp = qml.probs(wires=range(2))
+
+        result = measure_with_samples(mp, state, shots=shots)
+
+        if total_copies == 1:
+            assert isinstance(result, np.ndarray)
+            result = (result,)
+
+        assert isinstance(result, tuple)
+        assert len(result) == total_copies
+
+        for res in result:
+            assert res.shape == (4,)
+            assert res[0] == 0
+            assert res[3] == 0
+            assert res[1] + res[2] == 1
+
+    @pytest.mark.parametrize(
+        "shots, total_copies",
+        [
+            [(100,), 1],
+            [((100, 1),), 1],
+            [((100, 2),), 2],
+            [(100, 100), 2],
+            [(100, 200), 2],
+            [(100, 100, 200), 3],
+            [(200, (100, 2)), 3],
+        ],
+    )
+    def test_expval_measure_shot_vector(self, shots, total_copies):
```

Comment:
Not super necessary but might be nice to have a test for `var` as well.

Reply:
Added `var` tests 👍